framework/src/main/java/io/github/kgress/scaffold/BaseComponent.java

```diff
@@ -0,0 +1,234 @@
+              NoSuchMethodException e) {
+            throw new ComponentException(e);
+          }
+        });
+    return listOfComponents;
+  }
+
+  /**
+   * Converts {@link Field}'s from a class that extends off of {@link BaseComponent} from an
+   * "inaccessible" state to "accessible." We will only modify the access of Scaffold strongly typed
+   * elements. Afterwards, combines the parent and child together.
+   *
+   * @param componentInstance  the instance of the {@link BaseComponent}
+   * @param fullParentSelector the parent selector being used as the prefix
+   * @param <T>                the type reference of {@link BaseComponent}
+   */
+  private <T extends BaseComponent> void convertFieldsWithNewLocator(T componentInstance,
+      String fullParentSelector) {
+    var classFields = componentInstance.getClass().getDeclaredFields();
+
+    Arrays.stream(classFields).forEach(field -> {
+      var elementType = field.getGenericType();
+      try {
+        field.setAccessible(true);
+        BaseWebElement convertedElement = null;
+        if (elementType.getTypeName().contains("DivWebElement")) {
+          convertedElement = (DivWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("ButtonWebElement")) {
+          convertedElement = (ButtonWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("CheckBoxWebElement")) {
+          convertedElement = (CheckBoxWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("DateWebElement")) {
+          convertedElement = (DateWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("DropDownWebElement")) {
+          convertedElement = (DropDownWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("ImageWebElement")) {
+          convertedElement = (ImageWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("InputWebElement")) {
+          convertedElement = (InputWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("LinkWebElement")) {
+          convertedElement = (LinkWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("RadioWebElement")) {
+          convertedElement = (RadioWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("StaticTextWebElement")) {
+          convertedElement = (StaticTextWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("BaseWebElement")) {
+          convertedElement = (BaseWebElement) field.get(componentInstance);
+        } else {
+          log.debug(
+              "Scaffold detected a field during component list building that is not a defined as "
```

Comment:
If we're here, can we print the type name here to give a better message of what field isn't defined?

Reply:
Good suggestion, I will add that in the next commit